### ISSUE TITLE
[Feature Request] Create Members Manually

### ISSUE THREAD

**Issue by ItsNoted** [docmost/docmost#63]
The ability to create members in the back end without using email invites.

**Comment by Philipinho** [comment]
The idea is to also have a configurable workspace level invite link.
It could be `wiki.com/invites/join/workspace_invite_code`. 
This will not require sending a mail and anyone with the link can join.

**Comment by FromNetToDev** [comment]
Hello

> The idea is to also have a configurable workspace level invite link. It could be `wiki.com/invites/join/workspace_invite_code`. This will not require sending a mail and anyone with the link can join.

Hello, where find the workspace_invite_code?? where I should take this code?

**Comment by Philipinho** [comment]
@FromNetToDev , it’s not yet implemented.

**Comment by FromNetToDev** [comment]
What ways are there to add users, besides mail? I have difficulties with mail. Is it possible to connect to the database and create a user there? is there an instruction if such a scenario is possible?

**Comment by Philipinho** [comment]
@FromNetToDev you can find a workaround here https://github.com/docmost/docmost/discussions/37#discussioncomment-9940358.

**Comment by UrbanCircles** [comment]
any update on when this might become available?

**Comment by pwalther** [comment]
@johnnyitd you can get the invite link via the DB:

-U username -W dbName:
```
docker exec -it docmost-db-1 psql -U docmost -W docmost
select id, token, email from workspace_invitations;
```
Link is then: https://yourdomain.com/invites/${invitationId}?token=${inviteToken}
like: https://yourdomain.com//invites/01945c46-aa8e-7d75-b5ca-2d86a97f2840?token=mixbvk2mou2mznse

**Comment by LJAM96** [comment]
workaround works well to create users, would be nice if there was an ability to create offline links within the GUI for user setup.

Also results in password changes being very difficult.
@pwalther dont suppose you would know the correct SQL statements to pull the required data to manually access reset password requests

**Comment by pwalther** [comment]
I think this might work:

-U username -W dbName:

```
docker exec -it docmost-db-1 psql -U docmost -W docmost
select t.token from user_tokens t, users u where t.user_id = u.id and u.email like 'email@server.com';

```
 https://yourdomain.com/password-reset?token=${token}

**Comment by LJAM96** [comment]
> I think this might work:
> 
> -U username -W dbName:
> 
> ```
> docker exec -it docmost-db-1 psql -U docmost -W docmost
> select t.token from user_tokens t, users u where t.user_id = u.id and u.email like 'email@server.com';
> ```
> 
> [https://yourdomain.com/password-reset?token=${token}](https://yourdomain.com/password-reset?token=$%7Btoken%7D)

Works perfectly thank you

**Comment by k0lp4tzki** [comment]
Hi, on my search to create user locally /wo mail i stumbled across this topic and was happy that it's an actual thing.

I create users at this moment for me:

1. Generate a user with some kind of random email, because the mail is not more than a username (to me)
2. Without Mailing the invitation is stored in the database like described above.
3. because the invation link -with- configured mailing is referenced on my local network and not some public url, i prompted a script to generate invite link from the  workspace_invitations; table.

After a entry is just as long in the tabel as the invite is pending. So the outcome is good enough.


`#!/bin/bash
# Prompted by Dennis Kolpatzki (https://kolpatzki.de) and written by ChatGPT 4.0 
# because I'm lazy and need fast solutions to first-world problems.

# Database credentials
DB_USER="docmost"
DB_NAME="docmost"
DB_CONTAINER="docmost-db-1"
DB_PASSWORD="osEcKleMENchMAgrALDENcRE"  # Replace with actual password
DOMAIN="http://dyndns.kolpatzki.de:55000"

# Query to get all invitations
QUERY="SELECT id, token, email FROM workspace_invitations;"

# Execute the query and store the result in a variable
INVITATIONS=$(docker exec -i $DB_CONTAINER env PGPASSWORD=$DB_PASSWORD \
    psql -U $DB_USER -d $DB_NAME -A -t -F"," -c "$QUERY")

# Check if we got any data
if [[ -z "$INVITATIONS" ]]; then
    echo "No invitations found!"
    exit 1
fi

# Print and process all invitations
echo "Generated Invitation Links:"
echo "------------------------------------"
echo "$INVITATIONS" | while IFS=',' read -r ID TOKEN EMAIL; do
    # Remove unnecessary spaces
    ID=$(echo $ID | xargs)
    TOKEN=$(echo $TOKEN | xargs)
    EMAIL=$(echo $EMAIL | xargs)

    # Ensure values are valid before printing
    if [[ -n "$ID" && -n "$TOKEN" && -n "$EMAIL" ]]; then
        INVITE_LINK="$DOMAIN/invites/$ID?token=$TOKEN"
        echo "Invitation for: $EMAIL"
        echo "Link: $INVITE_LINK"
        echo "------------------------------------"
    fi
done

dockeradmin@dockersrv:/opt/docker/docmost$ cat get_links2.sh
#!/bin/bash

# Database credentials
DB_USER="docmost"
DB_NAME="docmost"
DB_CONTAINER="docmost-db-1"
DB_PASSWORD="SuperSecretPassword"  # Replace with actual password
DOMAIN="http://public-url-that-should-be-used-for-invite"

# Query to get all invitations
QUERY="SELECT id, token, email FROM workspace_invitations;"

# Execute the query and store the result in a variable
INVITATIONS=$(docker exec -i $DB_CONTAINER env PGPASSWORD=$DB_PASSWORD \
    psql -U $DB_USER -d $DB_NAME -A -t -F"," -c "$QUERY")

# Check if we got any data
if [[ -z "$INVITATIONS" ]]; then
    echo "No invitations found!"
    exit 1
fi

# Print and process all invitations
echo "Generated Invitation Links:"
echo "------------------------------------"
echo "$INVITATIONS" | while IFS=',' read -r ID TOKEN EMAIL; do
    # Remove unnecessary spaces
    ID=$(echo $ID | xargs)
    TOKEN=$(echo $TOKEN | xargs)
    EMAIL=$(echo $EMAIL | xargs)

    # Ensure values are valid before printing
    if [[ -n "$ID" && -n "$TOKEN" && -n "$EMAIL" ]]; then
        INVITE_LINK="$DOMAIN/invites/$ID?token=$TOKEN"
        echo "Invitation for: $EMAIL"
        echo "Link: $INVITE_LINK"
        echo "------------------------------------"
    fi
done
`

![Image](https://github.com/user-attachments/assets/00d36de9-06bf-4fcb-ac21-afdd43d08f39)
![Image](https://github.com/user-attachments/assets/cae2e6ec-8073-45f0-b0c0-b61f9f419fe0)

Mayb its helpful for some of you too

**Comment by k0lp4tzki** [comment]
Hi, on my search to create user locally /wo mail i stumbled across this topic and was happy that it's an actual thing.

I create users at this moment for me:

1. Generate a user with some kind of random email, because the mail is not more than a username (to me)
2. Without Mailing the invitation is stored in the database like described above.
3. because the invation link -with- configured mailing is referenced on my local network and not some public url, i prompted a script to generate invite link from the  workspace_invitations; table.

After a entry is just as long in the tabel as the invite is pending. So the outcome is good enough.

[https://github.com/k0lp4tzki/bash/blob/main/create_invites_docmost.sh](url)

![Image](https://github.com/user-attachments/assets/4c755811-c02a-42d4-acd7-88100790ff76)
![Image](https://github.com/user-attachments/assets/00d36de9-06bf-4fcb-ac21-afdd43d08f39)
![Image](https://github.com/user-attachments/assets/cae2e6ec-8073-45f0-b0c0-b61f9f419fe0)

Mayb its helpful for some of you too

**Comment by heinrihs-s** [comment]
If you are using portainer stack, this is how to obtain the link:


### Fast Way 🚀
```bash
docker exec -it docmost-db-1 psql -U docmost -W docmost -c "select id, token, email from workspace_invitations;"
```

### Step-by-Step Way
1. Open Portainer > Containers > `docmost-db-1` > Console tab
2. Run:
   ```bash
   psql -U docmost -W docmost
   ```
3. Then:
   ```sql
   select id, token, email from workspace_invitations;
   ```

### Build Your Invite Link
```
https://yourdomain.com/invites/${invitationId}?token=${inviteToken}
```

Example:
```
https://yourdomain.com/invites/01945c46-aa8e-7d75-b5ca-2d86a97f2840?token=mixbvk2mou2mznse
```

PS: You'll need the DB password for both methods.

**Comment by Philipinho** [comment]
Closing since you can now copy user invite links, starting from v0.8.4.

**Comment by zhanglun** [comment]
> > If you are using portainer stack, this is how to obtain the link:
> > 
> > 1. Open Portainer > Containers > `docmost-db-1` > Console tab
> > 2. Run and insert DB password:
> >    psql -U docmost -W docmost
> > 3. Then:
> >    select id, token, email from workspace_invitations;
> > 
> > ### Build Your Invite Link
> > ```
> > https://yourdomain.com/invites/${invitationId}?token=${inviteToken}
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Example:
> > ```
> > https://yourdomain.com/invites/01945c46-aa8e-7d75-b5ca-2d86a97f2840?token=mixbvk2mou2mznse
> > ```
> 
> make sure to first create an invite in the web GUI, else nothing will show here

You are right! 😄 But I used another method to create the user without using the webui. I manually inserted an invitation record in workspace_invitations.
